Integrations/GoogleCloudTranslate/GoogleCloudTranslate.py

```diff
@@ -0,0 +1,251 @@
+
+
+def test_module(client):
+    """
+    Returning 'ok' indicates that the integration works like it is supposed to. Connection to the service is successful.
+
+    Args:
+        client (Client): instance of the Client class
+
+    Returns:
+        'ok' if test passed, anything else will fail the test.
+    """
+
+    try:
+        client.get_supported_languages()
+        return 'ok'
+
+    except Exception as e:
+        return 'Test failed: {}'.format(str(e))
+
+
+def supported_languages(client):
+    """Returns the list of supported languages
+
+    Args:
+        client (Client): instance of the Client class
+
+    Returns:
+        The list of supported languages
+
+        readable_output (str): This will be presented in the war room - should be in markdown syntax - human readable
+        outputs (dict): Dictionary/JSON - saved in the incident context in order to be used as inputs for other tasks in the
+                 playbook
+        raw_response (dict): Used for debugging/troubleshooting purposes - will be shown only if the command executed with
+                      raw-response=true
+    """
+    result = client.get_supported_languages()
+
+    # readable output will be in markdown format - https://www.markdownguide.org/basic-syntax/
+    readable_output = 'Languages: {}'.format(', '.join([l['language_code'] for l in result]))
+    outputs = {
+        'GoogleCloudTranslate': {
+            'SupportedLanguages': result
+        }
+    }
+
+    return (
+        readable_output,
+        outputs,
+        result  # raw response - the original response
+    )
+
+
+def translate_text(client, args):
+    """Translates text
+
+    Args:
+        client (Client): instance of the Client class
+        args (dict): dictionary of arguments. The argument text is the text to
+            be translated, target is the ISO 2 letter code of the target language
+            (default: en), source is the ISO 2 letter code of the source language
+            (default: auto detect)
+
+    Returns:
+        The list of supported languages
+
+        readable_output (str): This will be presented in the war room - should be in markdown syntax - human readable
+        outputs (dict): Dictionary/JSON - saved in the incident context in order to be used as inputs for other tasks in the
+                 playbook
+        raw_response (dict): Used for debugging/troubleshooting purposes - will be shown only if the command executed with
+                      raw-response=true
+    """
+    text = args.get('text', '')
+    target = args.get('target', 'en')
+    source = args.get('source', None)
+
+    result = client.translate_text(
+        text,
+        target,
+        source=source
+    )
+
+    readable_output = 'Translation: {}\nSource Language Detected: {}'.format(
+        result['translated_text'],
+        result['detected_language_code']
+    )
+
+    outputs = {
+        'GoogleCloudTranslate': {
+            'TranslateText': {
```

Comment:
add DT to avoid duplicates: https://demisto.pan.dev/docs/dt
check by `text` and `source_language_code` (`detected_language_code`)

Reply:
I have added an ID to each request based on the text, source and target arguments. Used the ID attribute to dedup.